docs/Architecture Decision Record/019-zap-waiver.md

```diff
@@ -0,0 +1,24 @@
+# 1. Record architecture decisions
+
+Date: 2021-09-09 (yyyy-mm-dd)
+
+## Status
+
+Proposed
+
+## Context
+
+To satisfy ATO compliance we have sought to used OWASP's ZAP security scanner tool which displayed that we had not set a Content Security Policy (CSP). Upon implementing CSP headers to resolve this, we needed to take a compromise for style-src and utilize "unsafe inline" to allow use of the fedramped library 'uswds' as detailed in [the original ticket](https://github.com/raft-tech/TANF-app/issues/907) and [spike follow-up](https://github.com/raft-tech/TANF-app/issues/1208). 
+
+## Decision
+
+When investigating, we sought out information on what other agencies utilizing uswds had done in this situation. Another team had simply waived the requirement with the following justification: 
+
+> Weakness Description: Style-src unsafe-inline;: TTAHUB utilize javascript libraries to provide "typeahead search" functionality to filter select box options for end users. These libraries require setting the 'style' attribute on the HTML elements in a way that is incompatible with not including 'style-src: unsafe-inline' in our Content-Security-Policy (CSP) header.
+> Remediation Plan: This potential vulnerability is mitigated by utilizing React and JSX for generating HTML. JSX automatically escapes all user-supplied input)[sic] and thus is a compensating control for the CSP header.
```

Comment:
> JSX automatically escapes all user-supplied input)[sic] and thus is a

Typo?

Reply:
Yes, but as it is a quote, I left it in and noted the typo with "[sic]"